src/common.jl

```diff
@@ -1,194 +1,267 @@
+:qmax, :qmin, :qsteady_min, :qsteady_max, :qoldinit, :failfactor,
+:isoutofdomain, :unstable_check,
+:calck, :progress, :timeseries_steps, :dense)
 
 global warnlist = Set(warnkeywords)
 
 
 
-abstract type TaylorAlgorithm <: DiffEqBase.DEAlgorithm end
+abstract type TaylorAlgorithm <: OrdinaryDiffEqAdaptiveAlgorithm end
 struct TaylorMethod <: TaylorAlgorithm
     order::Int
 end
+struct _TaylorMethod <: TaylorAlgorithm
+    order::Int
+    parse_eqs::Bool
+end
+
+_TaylorMethod(order; parse_eqs=true) = _TaylorMethod(order, parse_eqs) # set `parse_eqs` to `true` by default
+
+alg_order(alg::TaylorMethod) = alg.order
+alg_order(alg::_TaylorMethod) = alg.order
+
+# Regarding `isfsal`, thought about setting isfsal to false, but low order
+# interpolation may be helpful (since Taylor interpolation is memory-intensive),
+# so let's set isfsal to true for now, which is the default, so no method
+# definition needed here, so we in principle should support FSAL
+# isfsal(::TaylorMethod) = true # see discussion above
 
 TaylorMethod() = error("Maximum order must be specified for the Taylor method")
 
 export TaylorMethod
 
-function DiffEqBase.solve(
-    prob::DiffEqBase.AbstractODEProblem{uType,tupType,isinplace},
-    alg::AlgType,
-    timeseries=[],ts=[],ks=[];
-    verbose=true,
-    saveat=eltype(tupType)[],
-    abstol = 1e-6,
-    save_start =  isempty(saveat) || saveat isa Number || prob.tspan[1] in saveat,
-    save_end   =  isempty(saveat) || saveat isa Number || prob.tspan[2] in saveat,
-    timeseries_errors=true, maxiters = 1_000_000,
-    callback=nothing, kwargs...) where
-        {uType, tupType, isinplace, AlgType <: TaylorAlgorithm}
-
-    tType = eltype(tupType)
-
-    if verbose
-        warned = !isempty(kwargs) && check_keywords(alg, kwargs, warnlist)
-        warned && warn_compat()
-    end
+# overload DiffEqBase.ODE_DEFAULT_NORM for Taylor1 arrays
+ODE_DEFAULT_NORM(x::AbstractArray{Taylor1{T}, N},y) where {T<:Number, N} = norm(x, Inf)
 
-    if haskey(prob.kwargs, :callback) || haskey(kwargs, :callback) || !isa(callback, Nothing)
-        error("TaylorIntegration is not compatible with callbacks.")
-    end
+### cache stuff
+struct TaylorMethodCache{uType, rateType, tTType, uTType} <: OrdinaryDiffEqMutableCache
+    u::uType
+    uprev::uType
+    tmp::uType
+    k::rateType
+    fsalfirst::rateType
+    tT::tTType
+    uT::uTType
+    duT::uTType
+    uauxT::uTType
+    parse_eqs::Ref{Bool}
+end
 
-    sizeu = size(prob.u0)
-    f = prob.f.f
+full_cache(c::TaylorMethodCache) = begin
+    tuple(c.u, c.uprev, c.tmp, c.k, c.fsalfirst, c.tT, c.uT, c.duT, c.uauxT, c.parse_eqs)
+end
 
-    tspan = prob.tspan
+# @show macroexpand(@__MODULE__, :(@cache struct TaylorMethodCache{uType,rateType,tTType,uTType} <: OrdinaryDiffEqMutableCache
+#   u::uType
+#   uprev::uType
+#   tmp::uType
+#   k::rateType
+#   fsalfirst::rateType
+#   tT::tTType
+#   uT::uTType
+#   duT::uTType
+#   uauxT::uTType
+#   parse_eqs::Bool
+# end) )
```

Comment:
Do we need this? Seems it is a left-over of something you were testing...

Reply:
😅 sorry, thanks! Yes, this was a leftover, it's now gone